paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVerticalLayoutUI.kt

```diff
@@ -78,14 +81,23 @@ internal fun PaymentMethodVerticalLayoutUI(
                 displayableSavedPaymentMethod = displayedSavedPaymentMethod,
                 resources = LocalContext.current.resources,
                 isEnabled = isEnabled,
-                isSelected = selectedIndex == -1,
+                isSelected = selection?.isSaved == true,
                 trailingContent = {
                     ViewMoreButton(onViewMorePaymentMethods = onViewMorePaymentMethods)
                 }
             )
             Text(stringResource(id = R.string.stripe_paymentsheet_new_pm), style = textStyle, color = textColor)
         }
 
+        val selectedIndex = remember(selection, paymentMethods) {
+            if (selection?.isSaved == false) {
```

Comment:
This would probably be easier to understand if it was written with the true branch first and it's also kind of hard to parse what happens if the selection is null. Is this accurate:
```
if (selection == null || selection.isSaved) {
 -1 
} else {
// find the code in payment methods
}
```